cmake/setup_cached_variables.cmake

```diff
@@ -129,6 +129,17 @@ IF( NOT "${CMAKE_BUILD_TYPE}" STREQUAL "Release" AND
 ENDIF()
 
 #
+# Force exactly one build type for multi-cofiguration generators.
+# Unfortunately CMake doesn't support custom configuration types so we just
+# call "DebugRelease" a "Release" build as well.
+#
+IF(CMAKE_BUILD_TYPE MATCHES "Release")
+  SET(CMAKE_CONFIGURATION_TYPES "Release" CACHE STRING "" FORCE)
+ELSE()
+  SET(CMAKE_CONFIGURATION_TYPES "Debug" CACHE STRING "" FORCE)
```

Comment:
It has a consequence in the sense that if you select "release" for example the target is compiled by concatenating general and config specific properties and variables. In order to support the simultaneous build of our debug and release library we happen to brutally remove all of these variables in ```cmake/setup_cached_variables.cmake```. This means that selecting in the GUI debug or release changes absolutely nothing.

This PR does not affect the testsuite (i.e., it doesn't fix the problem in the testsuite ^^) because the testsuite is configured as a subproject.